Java/syntax_test_java.java

```diff
@@ -1,25 +1,96 @@
 // SYNTAX TEST "Packages/Java/Java.sublime-syntax"
 
 package apple;
-// <- source.java meta.package.java keyword.other.package.java
-//      ^ meta.package.java support.other.package.java
-//           ^ meta.package.java punctuation.terminator.java
+// <- source.java meta.package-declaration.java keyword.other.package.java
+//      ^^^^^ entity.name.package.java
+//           ^ punctuation.terminator.java
+
+package com.example.apple;
+//^^^^^^^^^^^^^^^^^^^^^^^ meta.package-declaration.java
+//      ^^^^^^^^^^^^^^^^^ entity.name.package.java
+//         ^ punctuation.accessor.dot.java
+//                 ^ punctuation.accessor.dot.java
+//                       ^ punctuation.terminator.java
 
 import a.b.Class;
 // <- meta.import.java keyword.other.import.java
-//     ^ meta.import.java support.class.import.java
+//     ^^^^^^^^^ meta.path.java
+//     ^ support.type.package.java
+//      ^ punctuation.accessor.dot.java
+//       ^ support.type.package.java
+//        ^ punctuation.accessor.dot.java
+//         ^^^^^ support.class.import.java
+//              ^ punctuation.terminator.java
+
```

Comment:
Can you add a test for:

    import a.b.Class.SubClass;

SubClass gets drops out of the meta.import and meta.path scopes.

Reply:
I'll fix `meta.import`, but `meta.path` was intentional. Nested classes are members of the outer class so technically they can't be fully qualified. They just come along with the id of the outer class which can either be simple of fully qualified. Same logic is used in the other contexts as well.